fulcrumapp/fulcrum-query-sql
# Bug/flcrm 16235 fix filter linked column

# What?

This PR fixes an issue with filtering on linked record fields, allowing users to apply string-based filters on linked columns without returning empty results.

# Why?

Jira Ticket: [FLCRM-16235 | [web] - The number search in the record column filter of one app is not functioning as expected..](https://fulcrumapp.atlassian.net/browse/FLCRM-16235)

# Resolution

- In `src/ast/converter.js`, updated the `if` block handling `RecordLinkElements` within `whereClause` to fix filtering issues on linked records.
- Wrapped `ColumnRef` expressions in `AArrayExpr` to produce the `ANY(ARRAY[])` syntax, resolving the mismatch that was causing empty results. 

# Code Review
High quality code reviews are expected. You can utilize this [PR review checklist](https://fulcrumapp.atlassian.net/wiki/spaces/ENG/pages/1441693735/PR+Review+Checklist) to help you conduct one.

# Testing

- Use this preview [rails9316](https://rails9316.thefulcrum.team/admin) that has the `fulcrum-query-sql` dependancy pointing to the branch in this PR.
- Inside this preview, use the organization [1_Seeded Esri Organization. Safe to use!](https://rails9316.thefulcrum.team/admin/customers/57fcb303-998f-40da-92da-d4cf84ea9d67/members) that contains the same app where this issue was detected. [TFA0002 | Bus Shellter Installation](https://rails9316.thefulcrum.team/dash/bcea101f-ae35-4845-921f-6488b4e78435?mode=split&z=4&lat=71.145638&lon=146.104087)
- This app has two records with linked records (also included in the preview).
- Try filter them, to observe the expected behaviour.

# Video Demo


https://github.com/user-attachments/assets/32b9c7f9-845a-4967-b178-4a1d6e196c93
